js/post/update_post.js

```diff
@@ -428,4 +493,129 @@ function fillMisc() {
+ **/
+function fillCreditCount() {
+    'use-strict'
+
+    specCount = creditCountSpec  + creditCount300and400['Spec'];
+    majCount = creditCountMaj + creditCount300and400['Maj'];
+    minCount = creditCountMin + (creditCount300and400['Min'] * 0.5);
+
+
+    fillSpecCreditCount(specCount);
+    fillMajCreditCount(majCount);
+    fillMinCreditCount(minCount);
+}
+
+/**
+ * Autofills the credit count for Specialist
+ * @param {number} specCount The credit count for Specialist
+**/
+function fillSpecCreditCount(specCount) {
+    if (specCount >= 12) {
+        $('#spec_creds').html('(12.0/12.0)');
+    } else {
+        $('#spec_creds').html('(' + (specCount).toFixed(1) + '/12.0)');
+    }
+}
+
+/**
+ * Autofills the credit count for Major
+ * @param {number} majCount The credit count for Major
+**/
+function fillMajCreditCount(majCount) {
+    if (majCount >= 8) {
+        $('#maj_creds').html('(8.0/8.0)')
+    } else {
+        $('#maj_creds').html('(' + (majCount).toFixed(1) + '/8.0)');
+    }
+}
+
+/**
+ * Autofills the credit count for Minor.
+ * @param {number} minCount The credit count for Minor
+**/
+function fillMinCreditCount(minCount) {
+    if (minCount >= 4) {
+        $('#min_creds').html('(4.0/4.0)')
+    } else {
+        $('#min_creds').html('(' + (minCount).toFixed(1) + '/4.0)');
+    }
+}
+
+/**
+ * Autofills extra 200-level courses for last Minor constraint.
+ * @param {number} index The textbox number we are at
+ * @param HTMLElement} min300s Array of textbox elements to fill
+**/
+function addExtraMinCourses(index, min300s) {
+    for (var m = 0; m < 3; m++) {
+        if(index === 3) {
+            break;      
+        } if (getCookie(additional_min_200s[m]) === 'active') {
+            min300s[index].value = additional_min_200s[m];
+            min300s[index].readOnly = true;
+            creditCount300and400['Min'] += 0.5;
+            index += 1;
+        }
+        
+    }
+}
 
+/**
+ * Checks whether a POSt is completed and updates credit count colour if it is.
+**/
+function checkPostCompleted() {
+    if (categories_completed['Spec'] === 17) {
+        $('#spec_creds').css('color', 'green');
+    } else {
+        $('#spec_creds').css('color', 'red');
+    }
+    
+    if (categories_completed['Maj'] === 13) {
+        $('#maj_creds').css('color', 'green');
+    } else {
+        $('#maj_creds').css('color', 'red');
+    } 
+
+    if (categories_completed['Min'] === 6) {
+        $('#min_creds').css('color', 'green');
+    } else {
+        $('#min_creds').css('color', 'red');
+    }
+}
+
+/**
+ * Updates the Nav Bar on the Check My Post! page
+**/
+function updateNavPost() {
+    var nav_post = $('#nav-links')[0].getElementsByTagName('li')[3].getElementsByTagName('a')[0];
+
+    if (getCookie('specialist') === 'active') {
+        nav_post.innerHTML = 'Check My POSt! (' + (specCount).toFixed(1) + '/12.0)';
+        setCookie('activecount', (specCount).toFixed(1));
+    } if (getCookie('major') === 'active') {
```

Comment:
I haven't seen this style of if statements before. I don't know if it exists. In any case, please use style that is consistent with the rest of the project.